### ISSUE TITLE
Add support for endpoints requiring client authentication using PKI

### ISSUE THREAD

**Issue by cambriancoder** [huggingface/chat-ui#378]
Hi,

Are you open to adding support for endpoints that require client authentication using PKI? I have a requirement to use client authentication with our backend inference server. 

Currently authentication config from each endpoint is passed to the headers arg of the fetch command: https://github.com/huggingface/chat-ui/blob/main/src/lib/server/generateFromDefaultEndpoint.ts#L35

My quick googling has yielded this: https://sebtrif.xyz/blog/2019-10-03-client-side-ssl-in-node-js-with-fetch/ 
tl;dr; they create a `https.Agent(..)` which loads a PKI context from file which is passed to the `agent` arg in the fetch command. 

If you're happy for this to be added, how would you like to separate the logic of authentication using headers and client authentication using an SSL context?

Thank you! :)

**Comment by cambriancoder** [comment]
I have a PR here: https://github.com/huggingface/chat-ui/pull/393. tl;dr: optionally create a global mTLS context using client certificates. 

Node included a global `fetch` which conforms to the native fetch API. It's a wrapper around [Undici](https://undici.nodejs.org/#/) that has been imported into node-core since Node:18.
